Yasgui/packages/yasgui/src/Tab.ts

```diff
@@ -486,7 +522,10 @@ export class Tab extends EventEmitter {
     yasrConf.externalPluginsConfigurations = this.yasgui.config.yasr.externalPluginsConfigurations;
     yasrConf.downloadAsOptions = this.yasgui.config.yasr.downloadAsOptions;
 
-    this.yasr = new ExtendedYasr(this.yasqe, this.yasrWrapperEl, yasrConf, this.persistentJson.yasr.response);
+    if (!this.yasqe) {
+      throw Error("Yasqe is not initialized");
```

Comment:
The `yasqe` is created before `yasr`, and it can't be undefined in this line(`Yasr` can't be existed without `Yasqe`). I wrote this check because of TypeScript check, otherwise I have to add "Yasqe | undefined" everywhere in the code.
But you are right this code is confusing, I will change it.